db-service/lib/cqn4sql.js

```diff
@@ -991,7 +991,7 @@ function cqn4sql(originalQuery, model = cds.context?.model || cds.model) {
    */
   function getElementForRef(ref, def) {
     return ref.reduce((prev, res) => {
-      return prev?.elements?.[res] || prev?._target?.elements[res]
+      return (prev?.elements || prev?.foreignKeys)?.[res] || prev?._target?.elements[res] // PLEASE REVIEW: should we add the .foreignKey check here for the non-ucsn case?
```

Comment:
We don't need it, this function resolve also non-fk steps with `prev?._target?.elements[res]`, as far as I can tell, it shouldn't hurt to have it. We could then shortcut to the foreign key and not have to look into the assocs target.